src/Centreon/Infrastructure/Monitoring/TimelineRepositoryRDB.php

```diff
@@ -0,0 +1,430 @@
+
+        $collector->addValue(":hostId", $hostId);
+        $collector->addValue(":serviceId", (int) $serviceId);
+
+        // set ACL limitations
+        if (!$this->isAdmin()) {
+            $sql .= " INNER JOIN `:dbstg`.`centreon_acl` AS service_acl ON service_acl.host_id = s.host_id
+                  AND service_acl.service_id = s.service_id
+                  AND service_acl.group_id IN (" . $this->accessGroupIdToString($this->accessGroups) . ")";
+        }
+
+        //Group to avoid duplicate entries
+        $sql .= ' GROUP BY c.comment_id';
+
+        return $sql;
+    }
+
+    /**
+     * Generate SQL query for downtimes
+     *
+     * @param \Centreon\Infrastructure\CentreonLegacyDB\StatementCollector $collector
+     * @param int $hostId
+     * @param int $serviceId
+     * @return string
+     */
+    protected function generateDowntimesQuery(StatementCollector $collector, int $hostId, int $serviceId = null): string
+    {
+        $sql = "SELECT
+		CONCAT('D', d.downtime_id) AS `eventId`,
+        '". DowntimeEventObject::EVENTTYPE ."' AS `eventType`,
+        d.downtime_id AS `id`,
+		d.comment_data AS `output`,
+		d.entry_time AS `timestamp`,
+        NULL AS `status`,
+        d.type AS `type`,
+        NULL AS `retry`,
+        d.author AS `contact`,
+        NULL AS `command`,
+        NULL AS `persistent`,
+        d.deletion_time as `deletion_time`,
+        d.start_time AS `start_time`,
+        d.end_time AS `end_time`,
+        d.actual_start_time AS `actual_start_time`,
+        d.actual_end_time AS `actual_end_time`,
+        d.duration AS `duration`,
+        d.started AS `started`,
+        d.cancelled AS `cancelled`,
+        d.fixed AS `fixed`,
+        NULL AS `sticky`,
+        NULL AS `notify_contacts`
+        FROM `:dbstg`.`downtimes` d
+        WHERE d.host_id = :hostId AND d.service_id = :serviceId";
+
+        $collector->addValue(":hostId", $hostId);
+        $collector->addValue(":serviceId", (int) $serviceId);
+
+        // set ACL limitations
+        if (!$this->isAdmin()) {
+            $sql .= " INNER JOIN `:dbstg`.`centreon_acl` AS service_acl ON service_acl.host_id = s.host_id
+                  AND service_acl.service_id = s.service_id
+                  AND service_acl.group_id IN (" . $this->accessGroupIdToString($this->accessGroups) . ")";
+        }
+
+        //Group to avoid duplicate entries
+        $sql .= ' GROUP BY d.downtime_id';
+
+        return $sql;
+    }
+
+    /**
+     * Generate SQL query for ack
+     *
+     * @param \Centreon\Infrastructure\CentreonLegacyDB\StatementCollector $collector
+     * @param int $hostId
+     * @param int $serviceId
+     * @return string
+     */
+    protected function generateAcknowledgementsQuery(
+        StatementCollector $collector,
+        int $hostId,
+        int $serviceId = null
+    ): string {
+    
+        $sql = "SELECT
+		CONCAT('A', a.acknowledgement_id) AS `eventId`,
```

Comment:
IMHO, eventId should be an integer from only eventId
The unique id will be the association between eventType and eventId